zilliqa/tests/it/main.rs

```diff
@@ -1111,42 +1109,47 @@ fn compile_contract(path: &str, contract: &str) -> (Contract, Bytes) {
     });
 
     // Write the contract source to a file, so `solc` can compile it.
-    let mut contract_file = tempfile::Builder::new().suffix(".sol").tempfile().unwrap();
-    std::io::Write::write_all(&mut contract_file, &contract_source).unwrap();
-
-    let sc = ethers::solc::Solc::default();
+    let mut contract_file = tempfile::Builder::new()
+        .prefix("zq2")
+        .suffix(".sol")
+        .keep(true) // do not delete
+        .tempfile()
+        .unwrap();
+    contract_file.write_all(&contract_source).unwrap();
 
-    let mut compiler_input = CompilerInput::new(contract_file.path()).unwrap();
-    let compiler_input = compiler_input.first_mut().unwrap();
-    compiler_input.settings.evm_version = Some(EVM_VERSION);
+    let temp_path = contract_file.into_temp_path();
 
-    if let Ok(version) = sc.version() {
-        // gets the minimum EvmVersion that is compatible the given EVM_VERSION and version arguments
-        if EVM_VERSION.normalize_version(&version) != Some(EVM_VERSION) {
-            panic!(
-                "solc version {} required, currently set {}",
-                SHANGHAI_SOLC, version
-            );
-        }
-    }
+    let project = Project::builder()
+        .paths(
+            ProjectPathsConfig::hardhat(std::path::Path::new(env!("CARGO_MANIFEST_DIR"))).unwrap(),
+        )
+        .single_solc_jobs() // single file only
+        .locked_version(SolcLanguage::Solidity, semver::Version::new(0, 8, 26)) // downloads and installs, if not already in `svm list`
+        .build(Default::default())
+        .unwrap();
 
-    let out = sc
-        .compile::<CompilerInput>(compiler_input)
-        .unwrap_or_else(|e| {
-            panic!("failed to compile contract {}: {}", contract, e);
-        });
+    let output = project.compile_file(temp_path.to_path_buf()).unwrap();
 
-    // test if your solc can compile with v8.20 (shanghai) with
-    // solc --evm-version shanghai zilliqa/tests/it/contracts/Storage.sol
-    if out.has_error() {
-        panic!("failed to compile contract with error  {:?}", out.errors);
+    if output.has_compiler_errors() {
+        panic!(
+            "failed to compile contract with error  {:?}",
+            output.output().errors
+        );
     }
 
-    let contract = out
-        .get(contract_file.path().to_str().unwrap(), contract)
+    let contract = output
+        .output()
+        .contracts
+        .get(temp_path.to_path_buf().as_path(), contract)
         .unwrap();
+
     let abi = contract.abi.unwrap().clone();
     let bytecode = contract.bytecode().unwrap().clone();
+
+    // Work-around strict-typing
```

Comment:
Maybe this is clearer? Say no if you disagree :) 
```suggestion
    // Convert from the `alloy` representation of an ABI to the `ethers` representation, via JSON
```

Reply:
Done.